sycl/source/device.cpp

```diff
@@ -208,5 +208,78 @@ pi_native_handle device::getNative() const { return impl->getNative(); }
 
 bool device::has(aspect Aspect) const { return impl->has(Aspect); }
 
+void device::ext_oneapi_enable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextEnablePeerAccess>(Device,
+                                                                       Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+void device::ext_oneapi_disable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextDisablePeerAccess>(Device,
+                                                                        Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+bool device::ext_oneapi_can_access_peer(const device &peer,
+                                        ext::oneapi::peer_access attr) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+
+  if (Device == Peer) {
+    return true;
+  }
+
+  if (peer.get_backend() != backend::ext_oneapi_cuda) {
+    return false;
+  }
+
+  RT::PiPeerAttr PiAttr;
+  size_t return_size;
+  int value;
+
+  switch (attr) {
+  case ext::oneapi::peer_access::access_supported: {
+    PiAttr = PI_PEER_ACCESS_SUPPORTED;
+    break;
+  }
+  case ext::oneapi::peer_access::atomics_supported: {
+    PiAttr = PI_PEER_ATOMICS_SUPPORTED;
+    break;
+  }
+  }
```

Comment:
Just a small stylistic nit. Feel free to ignore.

```suggestion
  size_t return_size;
  int value;

  RT::PiPeerAttr PiAttr = [&]() {
    switch (attr) {
    case ext::oneapi::peer_access::access_supported:
      return PI_PEER_ACCESS_SUPPORTED;
    case ext::oneapi::peer_access::atomics_supported:
      return PI_PEER_ATOMICS_SUPPORTED;
    }
    throw sycl::exception(make_error_code(errc::invalid),
                          "Unrecognized peer access attribute.");
  }();
```